GitUI/BranchTreePanel/RepoObjectsTree.Nodes.Submodules.cs

```diff
@@ -263,9 +263,28 @@ private async Task<Nodes> LoadNodesAsync(SubmoduleInfoResult info, CancellationT
 
             private async Task LoadNodeDetailsAsync(CancellationToken token, Nodes loadedNodes)
             {
+                await TaskScheduler.Default;
+                foreach (var node in loadedNodes.DepthEnumerator<SubmoduleNode>())
+                {
+                    token.ThrowIfCancellationRequested();
+                    await node.LoadDetailsAsync(token).ConfigureAwaitRunInline();
+                }
+
+                await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Comment:
```suggestion
                await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync(token);
```

Reply:
In the line below already is `token.ThrowIfCancellationRequested();` and it can not be removed. We can pass the token here as well to emphasize that it is not enough to pass it here and we have to recheck if cancellation was requested.